src/Extensions/ComposerLoaderExtension.php

```diff
@@ -69,39 +67,29 @@ public function getPackages(array $allowedTypes = null)
 
     /**
      * Provides access to the Composer repository
-     *
-     * @return RepositoryInterface
      */
-    protected function getRepository()
+    protected function getRepository(): RepositoryInterface
     {
         /** @var Composer $composer */
         $composer = $this->getComposer();
-
-        /** @var BaseRepository $repository */
-        return new CompositeRepository([
-            new ArrayRepository([$composer->getPackage()]),
-            $composer->getRepositoryManager()->getLocalRepository(),
+        return new InstalledRepository([
+            new RootPackageRepository($composer->getPackage()),
+            $composer->getRepositoryManager()->getLocalRepository()
         ]);
     }
 
     /**
      * Find all dependency constraints for the given package in the current repository and return the strictest one
-     *
-     * @param BaseRepository $repository
-     * @param string $packageName
-     * @return string
      */
-    protected function getInstalledConstraint(BaseRepository $repository, $packageName)
+    protected function getInstalledConstraint(InstalledRepository $repository, string $packageName): string
```

Comment:
`Link::getPrettyConstraint()` can return null in composer 2.0.0
```suggestion
    protected function getInstalledConstraint(InstalledRepository $repository, string $packageName)
```

Reply:
I've added `$link->getPrettyConstraint() ?? ''` so that it's always a string